src/crccombine.c

```diff
@@ -0,0 +1,172 @@
+#include <stdint.h>
+#include <stdio.h>
+#include <strings.h>
+#include <immintrin.h>
```

Comment:
I suppose I didn't consider this originally, but I guess this isn't supported on ARM. I assume there is likely a version that implements. We could also just make this an x86 only optimization for now.

Reply:
Added some macros and a non-vector (but faster than bitwise) multiply function from the original patchset. Assuming everyone has _ARM_ defined as 0 or 1, this should now be cross-platform. (it exists on Linux x86 as 0)

I also bumped the cutoffs based on my experience with the slower combine functions.

Reply:
Valkey can be compiled for some weird systems so I'm not so sure we can assume `_ARM_` is always defined. (We have some ifdefs for Solaris and multiple \*BSDs.)

Better write `#if !defined(_ARM_) || !_ARM_` just in case, to avoid warnings when compiling with `-Wundef`.

Reply:
We defined `__arm__` in config.h, maybe we import that and use that define.

Reply:
@madolson I like the separation so i'd prefer to avoid including config.h if we can.

Reply:
Is it all that important? They are our files that we are modifying in the end.

Reply:
It's not that important but it seemed easy to avoid.

Reply:
It would hit us if we (or someone else) ever want to use this code for something else, like a separate rdb tool. That they're our files is not a good reason to add unnecessary coupling imo.

Reply:
I just realized that "serverassert.h" imports config.h, so if we want to be careful about writing portable code, we need to make sure to avoid the config.h defines since it will seemingly work until someone tries to do this replacement.

Reply:
Yes. :/ IMO it shouldn't. It's just for `likely` and `valkey_unreachable`, which only depend on the macro `__GNUC__`.

Reply:
Ok, going to close this since you proposed a fix here: https://github.com/valkey-io/valkey/pull/404.